From be3598d7737fa5ba4063a9374fb635a07a926400 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Fri, 24 Apr 2020 15:12:46 +0300
Subject: [PATCH 3/4] contrib: ffmpeg: Convert standalone patches to git format
 patches

This eases working with them as a complete set.

Also get rid of tabs from the videotoolbox patch.
---
 contrib/ffmpeg/A04-videotoolbox.patch | 25 +++++++++++++++++++------
 contrib/ffmpeg/A13-qsv-dx11.patch     | 22 ++++++++++++++++++++++
 2 files changed, 41 insertions(+), 6 deletions(-)

diff --git a/contrib/ffmpeg/A04-videotoolbox.patch b/contrib/ffmpeg/A04-videotoolbox.patch
index 387af0d66..575d78c53 100644
--- a/contrib/ffmpeg/A04-videotoolbox.patch
+++ b/contrib/ffmpeg/A04-videotoolbox.patch
@@ -1,27 +1,40 @@
+From 53b69d5f33c939a2221f85fea624256eb1a80431 Mon Sep 17 00:00:00 2001
+From: Damiano Galassi <damiog@gmail.com>
+Date: Thu, 15 Nov 2018 13:56:16 +0100
+Subject: [PATCH] videotoolboxenc: Handle a few more color primaries,
+ initialize has_b_frames for HEVC
+
+---
+ libavcodec/videotoolboxenc.c | 9 +++++++++
+ 1 file changed, 9 insertions(+)
+
 diff --git a/libavcodec/videotoolboxenc.c b/libavcodec/videotoolboxenc.c
-index 7796a68..e8b6245 100644
+index ff33c279c3..4551b548ea 100644
 --- a/libavcodec/videotoolboxenc.c
 +++ b/libavcodec/videotoolboxenc.c
-@@ -866,6 +866,14 @@ static int get_cv_color_primaries(AVCodecContext *avctx,
+@@ -877,6 +877,14 @@ static int get_cv_color_primaries(AVCodecContext *avctx,
              *primaries = NULL;
              break;
  
 +        case AVCOL_PRI_BT470BG:
-+			*primaries = kCVImageBufferColorPrimaries_EBU_3213;
++            *primaries = kCVImageBufferColorPrimaries_EBU_3213;
 +            break;
 +
 +        case AVCOL_PRI_SMPTE170M:
-+			*primaries = kCVImageBufferColorPrimaries_SMPTE_C;
++            *primaries = kCVImageBufferColorPrimaries_SMPTE_C;
 +            break;
 +
          case AVCOL_PRI_BT709:
              *primaries = kCVImageBufferColorPrimaries_ITU_R_709_2;
              break;
-@@ -1302,6 +1310,7 @@ static av_cold int vtenc_init(AVCodecContext *avctx)
+@@ -1313,6 +1321,7 @@ static int vtenc_configure_encoder(AVCodecContext *avctx)
          vtctx->get_param_set_func = compat_keys.CMVideoFormatDescriptionGetHEVCParameterSetAtIndex;
          if (!vtctx->get_param_set_func) return AVERROR(EINVAL);
          if (!get_vt_hevc_profile_level(avctx, &profile_level)) return AVERROR(EINVAL);
 +        vtctx->has_b_frames = avctx->max_b_frames > 0;
      }
  
-     vtctx->session = NULL;
+     enc_info = CFDictionaryCreateMutable(
+-- 
+2.17.1
+
diff --git a/contrib/ffmpeg/A13-qsv-dx11.patch b/contrib/ffmpeg/A13-qsv-dx11.patch
index 8da246073..43d016aaf 100644
--- a/contrib/ffmpeg/A13-qsv-dx11.patch
+++ b/contrib/ffmpeg/A13-qsv-dx11.patch
@@ -1,3 +1,22 @@
+From 465bfdb7a393819b023ffc79b50efd1827bb26a9 Mon Sep 17 00:00:00 2001
+From: Artem Galin <artem.galin@gmail.com>
+Date: Sat, 29 Feb 2020 12:31:23 +0000
+Subject: [PATCH] qsv: hw decoding improvements
+
+---
+ fftools/ffmpeg_opt.c          |  12 +-
+ libavcodec/qsv.c              |  69 ++++++++--
+ libavcodec/qsv_internal.h     |   9 ++
+ libavcodec/qsvdec.c           | 184 ++++++++++++++-------------
+ libavcodec/qsvdec.h           |   2 +
+ libavcodec/qsvdec_h2645.c     |   1 +
+ libavcodec/qsvdec_other.c     |   1 +
+ libavfilter/qsvvpp.c          |  36 ++++--
+ libavutil/hwcontext_d3d11va.c |  56 +++++++-
+ libavutil/hwcontext_qsv.c     | 233 ++++++++++++++++++++++++++++------
+ libavutil/hwcontext_qsv.h     |  11 +-
+ 11 files changed, 462 insertions(+), 152 deletions(-)
+
 diff --git a/fftools/ffmpeg_opt.c b/fftools/ffmpeg_opt.c
 index f5ca18aa64..75d9b5669b 100644
 --- a/fftools/ffmpeg_opt.c
@@ -1061,3 +1080,6 @@ index b98d611cfc..d21d58eb61 100644
  } AVQSVFramesContext;
  
  #endif /* AVUTIL_HWCONTEXT_QSV_H */
+-- 
+2.17.1
+
-- 
2.17.1

